osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringInstallationEndpoint.java

```diff
@@ -447,4 +470,72 @@ public DecoupleMbusDeviceByChannelResponse getDecoupleMbusDeviceByChannelRespons
         }
         return response;
     }
-}
+
+    @PayloadRoot(localPart = "SetSubscriptionInformationRequest", namespace = SMARTMETER_INSTALLATION_NAMESPACE)
+    @ResponsePayload
+    public SetSubscriptionInformationAsyncResponse setSubscriptionInformation(
```

Comment:
I'd like to check whether this ended up in the intended location in the GXF code base.
Looking at the story I see: "The IP-address must be saved to the OSGP core database via the GXF management SOAP service."
The place it ended up is the Smart Metering Installation SOAP service.

Given that the data that is managed is in the core `Device` entity, and not specific to the `SmartMeter` entity you could expect that this would end up in osgp-adapter-ws-core `DeviceManagementEndpoint`.

If it was intended to be smart meter only (could be, though a bit strange with the data in the general `Device`) there is still the `SmartMeteringManagementEndpoint` that looks it would fit better with the description in the story.

Reply:
after talking with our PO, the action was part of the installation process. so it is added there. I will recheck if this is the accepted location.